### PR TITLE
fix editBox end editing by pressing return on Win32 / MacOS

### PR DESCRIPTION
**Pull request by PPpro**
resolve: https://discuss.cocos2d-x.org/t/how-to-check-if-value-entered-in-to-the-editbox/51781

### CODE REVIEW

File: cocos/ui/edit-box/EditBox-mac.mm [hunk 1]
```diff
@@ -88,6 +88,9 @@ - (void)controlTextDidChange:(NSNotification *)notification {
     NSTextField *textField = [notification object];
     callJSFunc("input", [textField.stringValue UTF8String]);
 }
+- (void)controlTextDidEndEditing:(NSNotification *)obj {
+    cc::EditBox::complete();
+}
```
Comment (PPpro) [inline cocos/ui/edit-box/EditBox-mac.mm:93]: controlTextDidEndEditing would be called when textField loses focus or the key return is pressed

File: cocos/ui/edit-box/EditBox-mac.mm [hunk 2]
```diff
@@ -253,13 +255,6 @@ void init(const cc::EditBox::ShowInfo &showInfo) {
     init(showInfo);
 
     EditBox::_isShown = true;
-    EditBox::mouseDownListenerId = EventDispatcher::addCustomEventListener(EVENT_MOUSE_DOWN, EditBox::onMouseDown);
-}
```
Comment (PPpro) [inline cocos/ui/edit-box/EditBox-mac.mm:257]: no  need  to register the mouse  event any more

File: cocos/ui/edit-box/EditBox-win32.cpp [hunk 3]
```diff
@@ -159,7 +178,11 @@ void EditBox::show(const EditBox::ShowInfo &showInfo) {
                            NULL);
             std::wstring_convert<std::codecvt_utf8<wchar_t>> convert;
             CC_LOG_DEBUG("Can not create editbox: %s", convert.to_bytes(buffer).c_str());
+            return;
         }
+
+        g_prevMainWindowProc = (WNDPROC)SetWindowLongPtr(parent, GWL_WNDPROC, (LONG_PTR)mainWindowProc);
+        g_prevEditWindowProc = (WNDPROC)SetWindowLongPtr(g_hwndEditBox, GWL_WNDPROC, (LONG_PTR)editWindowProc);
```
Comment (PPpro) [inline cocos/ui/edit-box/EditBox-win32.cpp:185]: `g_prevMainWindowProc` can't receive the `WM_KEYUP` event

**Comment by PPpro** [comment]
@minggo please review

File: cocos/ui/edit-box/EditBox-mac.mm (hunk 2 above)
Comment (minggo) [reply cocos/ui/edit-box/EditBox-mac.mm:257]: Why no need it now?

Comment (minggo) [reply cocos/ui/edit-box/EditBox-mac.mm:257]: Got it, end editing event will cover it.

File: cocos/ui/edit-box/EditBox-mac.mm [hunk 4]
```diff
@@ -277,10 +272,6 @@ void init(const cc::EditBox::ShowInfo &showInfo) {
     }
 
     EditBox::_isShown = false;
-    if (EditBox::mouseDownListenerId != 0) {
-        EventDispatcher::removeCustomEventListener(EVENT_MOUSE_DOWN, EditBox::mouseDownListenerId);
-        EditBox::mouseDownListenerId = 0;
-    }
```
Comment (minggo) [inline cocos/ui/edit-box/EditBox-mac.mm:283]: So, can remove corresponding header files of EventDispatcher and related classes if have.

**Review by minggo — APPROVED** [review]

File: cocos/ui/edit-box/EditBox-mac.mm (hunk 1 above)
Comment (minggo) [reply cocos/ui/edit-box/EditBox-mac.mm:93]: Good to know.

**Pull request merged**